src/ashell/comms-uart.c

```diff
@@ -322,31 +303,18 @@ void comms_write_buf(const char *buf, int len)
     if (len == 0)
         return;
 
-    struct device *dev = dev_upload;
-    uart_irq_tx_enable(dev);
-
+    uart_irq_tx_enable(dev_upload);
     data_transmitted = false;
-    uart_fifo_fill(dev, buf, len);
-
+    uart_fifo_fill(dev_upload, (const uint8_t *)buf, len);
```

Comment:
I see you're right, some samples do exactly what you did, others ignore the return value just assume it works, without even the data transmitted loop.

The function @pfalcon pointed to shows that it will return 0 if there is no such api in the driver; so if that could ever happen we would hang in my above code. Maybe then 0 should be considered an error? Not sure, the samples and documentation are insufficient.

`test_uart_fifo.c` shows this in `test_fifo_fill`:
```c
		data_transmitted = false;
		while (!uart_fifo_fill(uart_dev, (uint8_t *) &fifo_data[i], 1));
		while (data_transmitted == false);
```

so I guess they're expecting 0 to be a recoverable error.

The only other sample code that ever checks the return value of the function is in `drivers/bluetooth/hci/h4.c` which considers it an error if it's not 1 (they were trying to write just one byte).

Maybe we should consider 0 an error for now to avoid hanging? Otherwise there doesn't seem to be any way to distinguish "the buffer's just full, I can't write any bytes" from "something's not working and will never work".